### ISSUE TITLE
Lift the model-agnostic half of `TypeSafeModel` into a `ClassifierModel` base

### ISSUE THREAD

**Issue by DouweM** [pydantic/pydantic-ai#8591]
### The observation

Almost nothing in [`TypeSafeModel`](https://pydantic.dev/docs/ai/models/typesafe/) is about Jev. Counting the current `models/typesafe.py`: **48 functions and methods, of which 10 lines touch `typesafe_sdk` at all.** The rest is a general answer to a question that is not Jev's — *how does a Pydantic AI output type become a set of typed questions, and how do the answers become an output?*

That general part is:

- the schema walk: an output type's fields to questions, and the refusals when a field cannot become one
- the three primitives themselves — a yes/no with a probability, a pick-one with a distribution, an ordered rubric
- choose-then-fill: one request picks a route, a second fills it (`_Ask`, added in #8589)
- thresholds, and what confidence means on `provider_details`
- everything a `Model` owes the agent loop: streaming as one event, message mapping, usage, `supports_text_output`

The Jev-specific part is: the SDK client, the mapping from our primitives to `Noul`/`Choice`/`Score`, and the published limits (255 options, 10 rubric levels).

### Why now

Open alternatives exist already. [`AlexWortega/openjev`](https://huggingface.co/AlexWortega/openjev) is MIT, ships 0.8B/4B/35B-MoE checkpoints, and serves over SGLang with an HTTP `/classify` endpoint — and we already have a vLLM provider for models in that family.

But it is **not** a Jev clone, and that is the useful part. It is an NLI cross-encoder: premise + hypothesis in, probabilities over `[contradiction, entailment, neutral]` out, plus a `rerank(question, options)` helper. The three primitives still map onto it:

| our primitive | on an NLI cross-encoder |
| --- | --- |
| yes/no | entailment probability of the field's question as a hypothesis |
| pick-one | `rerank` over the options, distribution from the scores |
| rubric | ordered hypotheses, probability-weighted position along them |

So the seam is **not** "the TypeSafe SDK". The seam is the three primitives. Anything that can answer them can be a classifier model.

### Proposed shape

```
Model
 └── ClassifierModel          # the ~80% above; abstract: ask(state, questions) -> answers
      ├── TypeSafeModel       # translates to typesafe_sdk, one request
      └── (OpenJevModel)      # generates hypotheses, calls /classify
```

`ClassifierModel` would be a subclass of `Model`, not a sibling. Branching `Model` into "LLM model" and "classifier model" is a much larger change, and `TypeSafeModel` has already shown that a non-text model can satisfy the existing `Model` contract without one — `supports_text_output` was the only profile flag it needed.

### The thing to design carefully

**The cost models are not the same, and the abstraction must not hide that.** A pick-one over 200 options is *one* request to Jev and *200* hypothesis evaluations on a cross-encoder. A rubric is one request to Jev and one per level elsewhere. If `ClassifierModel` presents both as "ask these questions", users will write code that is cheap on one backend and quadratic on another.

Whatever the interface ends up being, the number of underlying evaluations a question costs has to be visible to the implementation and, probably, reportable on `provider_details` the way `requests` already is for the two-request path.

Two smaller ones:

- **Limits are per-model.** 255 options and 10 rubric levels are Jev's; a cross-encoder has neither, and a different hosted model will have its own. These belong on the subclass, which is where [#8583](https://github.com/pydantic/pydantic-ai/pull/8583) has just put Jev's.
- **Not every backend returns calibrated probabilities.** Our confidence semantics and `typesafe_boolean_threshold` assume they are meaningful. Raw NLI logits are not calibrated the same way, and the independent evaluations collected in the community `awesome-jev` lists already show Jev itself is near-calibrated on public benchmarks but overconfident out of distribution.

### Not proposing to build this yet

One alternative is not enough to design an abstraction against; the right time is when we actually want to support a second. Filing so the shape is written down while the reasons are fresh, and so `TypeSafeModel` stops accumulating structure that would be awkward to lift later — #8589's `_Ask` is the first piece deliberately placed on the seam.
